app/models/allocation.rb

```diff
@@ -135,6 +137,21 @@ def dealloate_offender_after_transfer
     deallocate_offender event: Allocation::DEALLOCATE_PRIMARY_POM, event_trigger: Allocation::OFFENDER_TRANSFERRED if active?
   end
 
+  def last_version
+    papertrail_versions.last
+  end
+
+  # check for changes in the last week where the target value
+  # (item[1] in the array) is our staff_id
+  def new_case_for(staff_id)
+    papertrail_versions.where('created_at >= ?', 7.days.ago).map { |c|
+      YAML.load(c.object_changes)
+    }.select { |c|
+      c.key?('primary_pom_nomis_id') && c['primary_pom_nomis_id'][1] == staff_id ||
+        c.key?('secondary_pom_nomis_id') && c['secondary_pom_nomis_id'][1] == staff_id
+    }.any?
+  end
+
```

Comment:
Just for my own understanding, what is the benefit of moving this method out of allocated_offender model?

Reply:
Becuase it was only dependent on the 'allocation' object. This class is trying (along with several others) to work around our data model deficiencies, so it should probably not exist. This method is useful, so moving it into allocation is  one step towards removing the class entirely